Manager: Add migration note about react 18 (#25435)
storybookjs/storybook: MIGRATION.md

<h1>Migration</h1>

- [From version 7.x to 8.0.0](#from-version-7x-to-800)

>   - [Manager addons are now rendered with react 18](#manager-addons-are-now-rendered-with-react-18)

Author: ```suggestion
  - [Manager addons are now rendered with React 18](#manager-addons-are-now-rendered-with-react-18)
```

## From version 7.x to 8.0.0

> ### Manager addons are now rendered with react 18

Author: ```suggestion
### Manager addons are now rendered with React 18
```

> The UI added to the manager via addon, are now rendered with react 18.

Reviewer: @ndelangen when you have a moment can you adjust this sentence to be: 
"The UI added to the manager via addons is now rendered with React 18."

Also, can you capitalize react in the heading and adjust the link above accordingly?

> The UI added to the manager via addon, are now rendered with react 18.

Author: ```suggestion
The UI added to the manager via addons is now rendered with React 18.
```
